pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/LiteralTransformFunction.java

```diff
@@ -57,12 +57,20 @@
   public LiteralTransformFunction(String literal) {
     _literal = literal;
     _dataType = inferLiteralDataType(literal);
+    double mathLiteral;
     if (_dataType.isNumeric()) {
       BigDecimal bigDecimal = new BigDecimal(_literal);
       _intLiteral = bigDecimal.intValue();
       _longLiteral = bigDecimal.longValue();
       _floatLiteral = bigDecimal.floatValue();
       _doubleLiteral = bigDecimal.doubleValue();
+    } else if ((mathLiteral = getMathConstant(literal)) != 0) {
```

Comment:
@richardstartin is this approach fine or should I seperate this out entirely in a new class called `MathLiteralTransformFunctions` ?

Reply:
Blindly treating `E` and `PI` as number can cause unexpected behavior. They should be parsed within the math function context. Also I'd suggest addressing it in a separate PR.

Reply:
Ok. Yeah, I get for e.g. in a function like `startsWith(colA, 'E')` we should not be replacing `E` with Euler's consntat.